sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaCheckpointMark.java

```diff
@@ -35,21 +34,25 @@
  */
 @DefaultCoder(AvroCoder.class)
 public class KafkaCheckpointMark implements UnboundedSource.CheckpointMark {
+  private static final long OFFSET_DEDUP_PARTITIONS_PER_SPLIT = 1;
 
   private List<PartitionMark> partitions;
 
   @AvroIgnore
-  private Optional<KafkaUnboundedReader<?, ?>> reader; // Present when offsets need to be committed.
+  private KafkaUnboundedReader<?, ?> reader;
+
+  private boolean commitOffsetsInFinalize;
```

Comment:
@Abacn, I am not sure what the Avro stuff, such as AvroIgnore, means here. Perhaps commitOffsetsInFinalize should have AvroIgnore annotation as well?